src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs

```diff
@@ -1621,6 +1684,47 @@ public async Task Match_DisjunctionOverCounting(RegexEngine engine, string disju
             Assert.Equal(match, m.Value);
         }
 
+        public static IEnumerable<object[]> MatchAmbiguousRegexes_TestData()
+        {
+            // Different results in NonBacktracking vs Compiled or Interpreter
+            yield return new object[] { RegexEngine.NonBacktracking, "(a|ab|c|bcd){0,}d*", "ababcd", (0, 6) };
+            yield return new object[] { RegexEngine.NonBacktracking, "(a|ab|c|bcd){0,10}d*", "ababcd", (0, 6) };
+            yield return new object[] { RegexEngine.NonBacktracking, "(a|ab|c|bcd)*d*", "ababcd", (0, 6) };
+            yield return new object[] { RegexEngine.NonBacktracking, @"(the)\s*([12][0-9]|3[01]|0?[1-9])", "it is the 10:00 time", (6, 5) };
+
+            foreach (RegexEngine engine in new RegexEngine[] { RegexEngine.Compiled, RegexEngine.Interpreter })
```

Comment:
Same comment as earlier; I'd prefer to include all engines and then opt-out the ones to be excluded rather than the other way around so that, for example, we don't accidentally exclude the source generators.

Reply:
will take care of, as above, before merging